examples/camera-app/linux/src/clusters/webrtc_provider/webrtc-provider-manager.cpp

```diff
@@ -219,21 +218,54 @@ WebRTCProviderManager::ValidateStreamUsage(StreamUsageEnum streamUsage,
 void WebRTCProviderManager::ScheduleAnswerSend()
 {
     DeviceLayer::SystemLayer().ScheduleLambda([this]() {
-        CHIP_ERROR err = SendAnswerCommand(mCurrentSessionId, mSdpAnswer, mPeerId, mOriginatingEndpointId);
+        ChipLogProgress(NotSpecified, "Sending Answer command to node " ChipLogFormatX64, ChipLogValueX64(mPeerId.GetNodeId()));
 
-        if (err != CHIP_NO_ERROR)
-        {
-            ChipLogError(NotSpecified, "Answer failed: %" CHIP_ERROR_FORMAT, err.Format());
-        }
+        mCommandType = CommandType::kAnswer;
 
-        mOriginatingEndpointId = 0;
-        mOriginatingEndpointId = 0;
-        mPeerId                = ScopedNodeId();
+        // Attempt to find or establish a CASE session to the target PeerId.
+        CASESessionManager * caseSessionMgr = Server::GetInstance().GetCASESessionManager();
+        VerifyOrReturn(caseSessionMgr != nullptr);
+
+        caseSessionMgr->FindOrEstablishSession(mPeerId, &mOnConnectedCallback, &mOnConnectionFailureCallback);
     });
 }
 
-CHIP_ERROR WebRTCProviderManager::SendAnswerCommand(uint16_t sessionId, const std::string & sdpAnswer, const ScopedNodeId & peerId,
-                                                    EndpointId originatingEndpointId)
+void WebRTCProviderManager::OnDeviceConnected(void * context, Messaging::ExchangeManager & exchangeMgr,
+                                              const SessionHandle & sessionHandle)
+{
+    WebRTCProviderManager * self = reinterpret_cast<WebRTCProviderManager *>(context);
+    VerifyOrReturn(self != nullptr, ChipLogError(NotSpecified, "OnDeviceConnected:: context is null"));
+
+    ChipLogProgress(NotSpecified, "CASE session established, sending command with Command Type: %d...",
+                    static_cast<int>(self->mCommandType));
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    switch (self->mCommandType)
+    {
+    case CommandType::kAnswer:
+        err = self->SendAnswerCommand(exchangeMgr, sessionHandle);
```

Comment:
For sending this Answer command back, I guess the CommandPathParams need to be set with the EndpointId that was set by the initiating ProvideOffer from the client?

Reply:
Yes, it is set from the value of incoming command https://github.com/project-chip/connectedhomeip/blob/master/examples/camera-app/linux/src/clusters/webrtc_provider/webrtc-provider-manager.cpp#L108.

We have to pass this endpoint via class member variable mOriginatingEndpointId since this is triggered from callback